### PR TITLE
feat: Add a resource to manage sequences

### PR DESCRIPTION
**Pull request by dfedde**
Addresses feature-request #509 

## Test Plan

* [x] acceptance tests
* [x] used terraform to run test creates, updates, destroys and fixing changes made outside of terraform.
* [x] has been run and is working in rxrevu's infrastructure

## References
* https://docs.snowflake.com/en/sql-reference/sql/create-sequence.html

### CODE REVIEW

File: pkg/resources/sequence.go [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+		Description: "The database in which to create the sequence. Don't use the | character.",
+	},
+	"schema": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The schema in which to create the sequence. Don't use the | character.",
+	},
+	"next_value": {
+		Type:        schema.TypeInt,
+		Description: "The next value the sequance will provide.",
```
Comment (alldoami) [inline pkg/resources/sequence.go:44]:
```suggestion
		Description: "The next value the sequence will provide.",
```

**Comment by alldoami** [comment]
I'm not totally familiar with sequences, but should we try and provide a `start` optional parameter?

**Comment by dfedde** [comment]
We could add a start but it is a little confusing on how to implement it. It is only useful on create. On an update we don't want to regress the counter back to its start. We would also need to be smart in the cases where the sequence is recreated.

I was unable to figure out a way to implement it cleanly and was not useful for our use case so I left it out.

**Comment by alldoami** [comment]
@dfedde cool, wanna update your branch and commit the change I suggested above?

**Review by alldoami — APPROVED** [review]

**Pull request merged**